src/main/java/com/recipapp/service/AbstractDocumentService.java

```diff
@@ -0,0 +1,27 @@
+package com.recipapp.service;
+
+import com.recipapp.model.AbstractDocument;
+import com.recipapp.model.SequenceIdentifier;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+public abstract class AbstractDocumentService<O extends AbstractDocument, I> {
+
+    @Autowired
+    private SequenceService sequenceService;
+
+    public O save(O document) {
+        O modifiedDoc = this.preCreation(document);
+        if (modifiedDoc.getId() == null) {
```

Comment:
Is there any case where modifiedDoc.getId() this isn't going to be null?

Reply:
when you are updating a document